ci/ansible/roles/pulp-restore-backup/tasks/main.yaml

```diff
@@ -0,0 +1,99 @@
+      - local_bkup_dir is defined
+    msg: |
+      Location of remote backup directory and local directory for backup to be copied to
+      must be specified with varibales remote_bkup_dir and local_bkup_dir
+
+- name: Install dependencies
+  action: "{{ ansible_pkg_mgr }} name={{ item }} state=latest"
+  with_items:
+    - tar
+    - python34
+    - python34-setuptools
+    - gcc
+
+- name: Stop services
+  service: "name={{ item }} state=stopped"
+  with_items:
+    - httpd
+    - pulp_celerybeat
+    - pulp_resource_manager
+    - pulp_streamer
+    - pulp_workers
+    - qpidd
+
+- block:
+    - name: Create temporary folder to copy backups into
+      file: "path={{ remote_bkup_dir }} state=directory"
+
+    - name: Copy filesystem backups to remote
+      copy:
+        src: "{{ local_bkup_dir }}/{{ item.value.archive }}"
+        dest: "{{ remote_bkup_dir }}"
+      with_dict: "{{ files_to_restore }}"
+
+    - name: Unarchive filesystem backups
+      command: "tar vxjf {{ remote_bkup_dir }}/{{ item.value.archive }} --overwrite"
+      args:
+        chdir: "/"
+      with_dict: "{{ files_to_restore }}"
+
+    - name: Reset SELinux labels
+      command: "restorecon -r {{ item.value.dest_dir }}"
+      with_dict: " {{ files_to_restore }}"
+
+    - name: Drop pulp database
+      command: "mongo pulp_database --eval 'db.dropDatabase()'"
+
+    # unarchive module fails probably because
+    # it relies on the using the "--diff" option which fails when
+    # unarchiving a directory full of files that don't yet exist
+    # See: https://github.com/ansible/ansible-modules-core/issues/2936#issuecomment-189579092
+    - name: Copy database backups to remote
+      copy:
+        src: "{{ local_bkup_dir }}/{{ item.value.archive }}"
+        dest: "{{ remote_bkup_dir }}"
+      with_dict: "{{ db_to_restore }}"
+
+    - name: Unarchive database backups
+      command: "tar vxjf {{ remote_bkup_dir }}/{{ item.value.archive }}"
+      args:
+        chdir: "{{ remote_bkup_dir }}"
+      with_dict: "{{ db_to_restore }}"
+
+    - name: Restore database from backup
+      command: "mongorestore {{ remote_bkup_dir }}/{{ db_to_restore.mongodb.dir }}"
+
+    - name: Run pulp-manage-db
+      command: "sudo -u apache pulp-manage-db"
+
+    - name: Delete temp backup folder from server
+      file: path={{ remote_bkup_dir }} state=absent
+
+- name: Start services
+  service: "name={{ item }} state=started"
+  with_items:
+    - httpd
+    - pulp_celerybeat
+    - pulp_resource_manager
+    - pulp_streamer
+    - pulp_workers
+    - qpidd
+
+- block:
+
+  - name: Create config directory
+    file: path="{{ ansible_user_dir }}/.config/pulp_smash" state=directory
+
+  - name: Ensure a Pulp Smash configuration file is present
+    template:
+      src: settings.j2
+      dest: "{{ ansible_user_dir }}/.config/pulp_smash/settings.json"
+
+  - name: Test that content is restored
+    script: test_restore.sh "{{ test_results_dir }}"
```

Comment:
I'm guessing that this script can be dropped, much like the populate script can be dropped.